pyop2/base.py

```diff
@@ -2980,6 +3143,8 @@ class DecoratedMap(Map, ObjectCached):
 
     def __new__(cls, map, iteration_region=None, implicit_bcs=None,
                 vector_index=None):
+        if map is None:
+            return None
```

Comment:
Well if your map is None (because you are a Global), any corresponding DecoratedMap will also be None.